api/src/opentrons/protocol_engine/execution/movement.py

```
@@ -134,15 +145,28 @@ async def save_position(
             else:
                 pip_cp = CriticalPoint.NOZZLE
 
-        # TODO (spp, 2021-11-3): Handle MustHomeError case
-        point = await self._hardware_api.gantry_position(
-            mount=hw_mount,
-            critical_point=pip_cp,
-        )
+        try:
+            point = await self._hardware_api.gantry_position(
+                mount=hw_mount,
+                critical_point=pip_cp,
+            )
+        except HardwareMustHomeError as e:
+            raise MustHomeError(str(e)) from e
```

This seems to be swallowing the error. The command succeeds with some gibberish (I think) X & Y values in the response.

---

I think the problem was that `hardware_api.gantry_position` would actually never throw a `MustHomeError`, for the same reason as `hadware_api.move_rel`. Applied the same fix for `gantry_position` as #8724, can you check if this resolves the issue?